### PR TITLE
boot: Miscellaneous cleanups

### PR DESCRIPTION
**Pull request by medhefgo**

### CODE REVIEW

File: src/boot/efi/linux.c [hunk 1]
```diff
@@ -16,7 +16,7 @@
 #include "pe.h"
 #include "util.h"
 
-static EFI_LOADED_IMAGE * loaded_image_free(EFI_LOADED_IMAGE *img) {
+EFI_LOADED_IMAGE_PROTOCOL *loaded_image_free(EFI_LOADED_IMAGE_PROTOCOL *img) {
```
Comment (yuwata) [inline src/boot/efi/linux.c]: Why 'static' dropped??

**Review by yuwata — COMMENTED** [review]
Looks mostly good. One question.

File: src/boot/efi/linux.c (hunk 1 above)
Comment (medhefgo) [reply src/boot/efi/linux.c]: It's not needed (everywhere else we use the typedef version anyway).

Comment (medhefgo) [reply src/boot/efi/linux.c]: nvm, by brain saw this as a `struct`.

**Review by yuwata — APPROVED** [review]
LGTM.

**Pull request merged**